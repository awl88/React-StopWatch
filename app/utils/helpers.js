import { isPlainObject } from 'lodash'

const ERROR_DEFAULT = 'An error occurred. Please try again.'
const ERROR_NO_INTERNET = 'Request could not be made. Please check your internet connection.'

export const createWithDoc = ({ envName = '', docFunction = () => {}, component = '' }) => {
  let createComponentWithDoc
  if (envName !== 'production') {
    createComponentWithDoc = docFunction(component) // eslint-disable-line global-require
  }
  return createComponentWithDoc || component
}

export const getErrorMessage = err => {
  // - Axios does not return a response object if the service cannot be reached
  if (!err.response) {
    return ERROR_NO_INTERNET
  }

  // - If we do have a response and data object, use the full error object
  if (err.response && err.response.data) {
    try {
      const errorStringOrObject = Object.values(err.response.data)[0]
      if (isPlainObject(errorStringOrObject)) return errorStringOrObject[0]
      return errorStringOrObject
    } catch (error) {
      // eslint-disable-next-line no-console
      console.warn('Could not infer error message from error response object')
    }
  }

  // If a message has not been set, fallback to our default message
  return ERROR_DEFAULT
}

/**
 * Updates the state object with the updated field.
 * @param {object} state
 * @param {object} updatedField
 */
export const updateState = (state, updatedField) => ({
  ...state,
  ...updatedField,
})

/**
 * Updates an array stored in state.
 * - `add`: adds value to the array
 * - `remote`: removes the value at the index from the array
 * - `update`: updates the value at the index in the array
 * - `reset` and `filter`: returns the supplied value
 * - `clear`: removes all values from the array
 * @param {object} state
 * @param {object} updatedState
 */
export const updateArrayState = (state, { type, value, index }) => {
  switch (type) {
    case 'add':
      return [...state, value]
    case 'remove': {
      const updatedState = [...state]
      updatedState.splice(index, 1)
      return updatedState
    }
    case 'update': {
      const updatedState = [...state]
      updatedState[index] = value
      return updatedState
    }
    case 'reset':
    case 'filter':
      return value
    case 'clear':
      return []
    default:
      return state
  }
}
