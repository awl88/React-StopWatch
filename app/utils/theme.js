import colors from './colors'

const theme = {
  global: {
    font: {
      family: 'Open Sans',
    },
    colors: {
      brand: colors.primary,
    },
  },
}

export default theme
