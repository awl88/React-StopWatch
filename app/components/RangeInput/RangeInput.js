/* eslint-disable react/prop-types */
import React from 'react'

// Components
import { RangeInput as GrommetRangeInput } from 'grommet'

// Helpers
import { doc } from './RangeInput.doc'
import { createWithDoc } from '../../utils/helpers'

/**
 *
 * RangeInput
 *
 */
const RangeInput = ({ children, ...rest }) => (
  <GrommetRangeInput {...rest}>{children}</GrommetRangeInput>
)

export default createWithDoc({
  envName: process.env.NODE_ENV,
  docFunction: doc,
  component: RangeInput,
})
