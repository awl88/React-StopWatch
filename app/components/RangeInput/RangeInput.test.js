import React from 'react'
import { render } from '@testing-library/react'
// import ''@testing-library/jest-dom/extend-expect' // add some helpful assertions

// Components
import { TestWrapper } from 'utils/TestWrapper'
import RangeInput from './RangeInput'

const renderComponent = (props = {}) =>
  render(
    <TestWrapper>
      <RangeInput {...props} />
    </TestWrapper>,
  )

/**
 *
 * Tests for RangeInput
 *
 */
describe('RangeInput', () => {
  it('renders and matches snapshot', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
