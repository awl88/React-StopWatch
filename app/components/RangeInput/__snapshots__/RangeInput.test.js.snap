// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeInput renders and matches snapshot 1`] = `
<div>
  <div
    class="StyledGrommet-sc-19lkkz7-0 fUxOqw"
  >
    <input
      class="StyledRangeInput-sc-15st9ck-0 dkEJvl"
      type="range"
      value=""
    />
  </div>
</div>
`;
